### PR TITLE
Still more noderefactor

### PR DESCRIPTION
**Pull request by samchrisinger**
# Purpose/Changes

- prevent category selection from OB
- let update logs show changes
- fix breaking test
- use correct URL in PO search for existing projects
- Fix: https://trello.com/c/95ZMue06/92-read-only-child-of-private-project-is-not-listed-in-project-organizer

# Side Effects

None

### CODE REVIEW

File: website/views.py [hunk 1]
```diff
@@ -169,7 +169,7 @@ def get_all_registrations_smart_folder(auth, **kwargs):
     ).sort('-title')
 
     keys = nodes.get_keys()
-    return [rubeus.to_project_root(node, auth, **kwargs) for node in nodes if node.ids_above.isdisjoint(keys)]
+    return [rubeus.to_project_root(node, auth, **kwargs) for node in nodes if {node.parent_id}.isdisjoint(keys)]
```
Comment (sloria) [inline website/views.py]: Could you just do 'node.parent_id in keys`?

Comment (samchrisinger) [inline website/views.py]: Semantically I like isdisjoint better, but yes.

File: website/project/model.py [hunk 2]
```diff
@@ -976,10 +977,19 @@ def update(self, fields, auth=None, save=True):
...
                          'node': self._id,
-                         'updated_fields': list(updated)
+                         'updated_fields': {
+                             key: {
+                                 'old': values[key]['old'],
+                                 'new': values[key]['new']
+                             }
+                             for key in values
+                         },
+                         'updated_fields_display': ["{0} to {1}".format(key, Node.CATEGORY_MAP[values[key]['new']]) for key in values],
```
Comment (sloria) [inline website/project/model.py]: This is probably something the client code should do.

**Comment by sloria** [comment]
Pass finished

**Comment by sloria** [comment]
osf___conductivity_hydrogen_fauna_movement_consistent_interdependence_alignment_settings

It might make sense to hide the Change and Cancel buttons for top-level projects (but keep the disabled select input); it's a little hard to tell that the buttons are disabled.

**Comment by sloria** [comment]
I can make the above change when I merge.

**Comment by sloria** [comment]
Actually, there are other changes to make. @samchrisinger Can you make the above change?

**Comment by sloria** [comment]
I can't change the category of a component:

osf___child_settings

File: website/templates/log_templates.mako [hunk 3]
```diff
@@ -95,8 +95,11 @@ edited description of  <a class="log-node-title-link" data-bind="attr: {href: no
 </script>
 
 <script type="text/html" id="updated_fields">
-updated the <span data-bind="listing: params.updated_fields"></span> of 
-<a class="log-node-title-link" data-bind="attr: {href: nodeUrl}, text: nodeTitle"></a>
+  changed the <span data-bind="listing: {
+                                 data: params.updated_fields,
+                                 map: function(key, item) { return key + ' to ' + item.new;}
```
Comment (sloria) [inline website/templates/log_templates.mako]: This displays the internal representation for categories, not the user-friendly representations.

**Pull request merged**